app/src/main/java/space/taran/arknavigator/ui/fragments/FoldersFragment.kt

```diff
@@ -85,86 +65,76 @@ class FoldersFragment: MvpAppCompatFragment(), FoldersView, BackButtonListener {
         Log.d(FOLDERS_SCREEN, "view created in FoldersFragment")
         super.onViewCreated(view, savedInstanceState)
         App.instance.appComponent.inject(this)
-        initialize()
     }
 
-    override fun onResume() {
-        Log.d(FOLDERS_SCREEN, "resuming in FoldersFragment")
-        super.onResume()
-        presenter.resume()
-    }
+    override fun init() {
+        Log.d(FOLDERS_SCREEN, "initializing FoldersFragment")
+        (activity as MainActivity).setSelectedTab(0)
+        (activity as MainActivity).setToolbarVisibility(false)
 
-    override fun backClicked(): Boolean {
-        Log.d(FOLDERS_SCREEN, "[back] clicked in FoldersFragment")
-        return presenter.quit()
+        fab_add_roots.setOnClickListener {
+            presenter.onAddRootBtnClick()
+        }
     }
 
-    override fun notifyUser(message: String, moreTime: Boolean) {
-        Notifications.notifyUser(context, message, moreTime)
+    override fun updateFoldersTree(devices: List<Path>, folders: Folders) {
+        foldersTree = FoldersTree(devices, folders, presenter::onFoldersTreeAddFavoriteBtnClick, router)
+        rv_roots.adapter = foldersTree
     }
 
+    override fun updateRootPickerDialogPath(path: Path) {
+        folderPicker?.updatePath(path)
+    }
 
-    private fun initialize() {
-        Log.d(FOLDERS_SCREEN, "initializing FoldersFragment")
-
-        devices = listDevices(requireContext())
-
-        (activity as MainActivity).setSelectedTab(0)
-        (activity as MainActivity).setToolbarVisibility(false)
+    //todo fake disabling (still show messages when pressing on disabled button)
+    //todo consistent rules for onPick messages and gray-out
+    //todo revert button state when backClicked
```

Comment:
I've created https://github.com/ARK-Builders/ARK-Navigator/issues/64 instead of these 3 `todo`s, so they can be removed now from the code.